swan-lake/learn/mocking.md

```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
-
-    // stubbing to return a value for any calls to `get` function
-	test:prepare(clientEndpoint).when("get").thenReturn(getMockResponse());
-    // stubbing to return a value based on input
+    // stub to return the specified mock response when `get` function is called
+    test:prepare(clientEndpoint).when("get").thenReturn(getMockResponse());
+    // stub to return the specified mock response when the specified argument is passed
     test:prepare(clientEndpoint).when("get").withArguments("/categories").thenReturn(getCategoriesResponse());
-    
-    string result = checkpanic getRandomJoke("Sheldon");
-    test:assertEquals(result, "When Sheldon wants an egg, he cracks open a chicken.");
+    // invoke the function to test
+    string result = checkpanic getRandomJoke("Sanjiva");
+    // verify the return value against the expected string
+    test:assertEquals(result, "When Sanjiva wants an egg, he cracks open a chicken.");
 }
 
-// stubbing to return different values for each function call
+// This test stubs the behavior of the `get` function to return a specified sequence of values
+// for each `get` function invocation  when testing the `getRandomJoke` function
 @test:Config {}
 public function testReturnSequence() {
-	clientEndpoint = <http:Client>test:mock(http:Client);
-	test:prepare(clientEndpoint).when("get").thenReturnSequence(getCategoriesResponse(), getMockResponse());
-
-	string result = checkpanic getRandomJoke("Sheldon");
-	test:assertEquals(result, "When Sheldon wants an egg, he cracks open a chicken.");
-
+    // create a default mock HTTP Client and assign it to the `clientEndpoint` object
+    clientEndpoint = <http:Client>test:mock(http:Client);
+    // stub to return the corresponding value for each invocation. i.e. the first call to `get` will return the mock
+    // response containing categories and the second call will return the mock response containing the joke
+    test:prepare(clientEndpoint).when("get").thenReturnSequence(getCategoriesResponse(), getMockResponse());
+    // invoke the function to test
+    string result = checkpanic getRandomJoke("Sanjiva");
+    // verify the return value against the expected string
+    test:assertEquals(result, "When Sanjiva wants an egg, he cracks open a chicken.");
 }
 
-// stubbing a member variable value
+// This function shows how to stub a member variable value of the HTTP Client object
 @test:Config {}
 function testMemberVariable() {
+    // create a default mock HTTP Client and assign it to the `clientEndpoint` object
     clientEndpoint = <http:Client>test:mock(http:Client);
+    // stub the value of `url` variable to return the specified string
     test:prepare(clientEndpoint).getMember("url").thenReturn("https://foo.com/");
+    // verify the specified value is set
     test:assertEquals(clientEndpoint.url, "https://foo.com/");
 }
 
-// util function for constructing a mock response
+// Returns a mock HTTP response to be used for random joke API invocation
```

Comment:
```suggestion
// Returns a mock HTTP response to be used for random joke API invocation
```
If we are starting with a capital letter please do the same in all comments for consistency.

Reply:
Capital letters are used for only function descriptions here.

Reply:
ok